File: apps/02-personal-md-files/backend/python/src/main.py
import os
from dotenv import load_dotenv
from fastapi import FastAPI, HTTPException, Request
from fastapi.middleware.cors import CORSMiddleware
from fastapi.responses import JSONResponse
from src.controllers.chat_controller import ChatController
from src.services.llm_service import LLMService
from src.services.markdown_service import MarkdownService

# Load environment variables from .env file
load_dotenv()

# Create FastAPI app
app = FastAPI(title="Simple Chat Backend")

# Configure CORS
app.add_middleware(
    CORSMiddleware,
    allow_origins=os.getenv("CORS_ALLOWED_ORIGINS", "http://localhost:3001").split(","),
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)

# Initialize services
llm_service = LLMService()
markdown_service = MarkdownService()
chat_controller = ChatController(llm_service, markdown_service)

@app.get("/")
async def root():
    """Root endpoint"""
    return {"message": "Personal MD Files Backend"}

@app.post("/api/chat")
async def chat(request: Request):
    """Chat endpoint"""
    try:
        data = await request.json()
        result = await chat_controller.chat(data)
        return result
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

@app.get("/api/providers")
async def providers():
    """Get available providers endpoint"""
    try:
        result = chat_controller.providers()
        return result
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

@app.post("/api/switch-provider")
async def switch_provider(request: Request):
    """Switch provider endpoint"""
    try:
        data = await request.json()
        result = chat_controller.switch_provider(data)
        return result
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

if __name__ == "__main__":
    import uvicorn
    port = int(os.getenv("PORT", 8003))
    uvicorn.run("main:app", host="0.0.0.0", port=port, reload=True) 

File: apps/02-personal-md-files/backend/golang/src/main.go
package main

import (
	"fmt"
	"log"
	"net/http"
	"os"
	"strings"

	"github.com/gin-contrib/cors"
	"github.com/gin-gonic/gin"
	"github.com/joho/godotenv"
	"github.com/smoogie/rag_demo/apps/02-personal-md-files/backend/golang/src/controllers"
	"github.com/smoogie/rag_demo/apps/02-personal-md-files/backend/golang/src/services"
)

func main() {
	// Load environment variables from .env file
	if err := godotenv.Load(); err != nil {
		log.Printf("Warning: Error loading .env file: %v", err)
	}

	// Initialize services
	llmService := services.NewLLMService()
	markdownService := services.NewMarkdownService()
	chatController := controllers.NewChatController(llmService, markdownService)

	// Create Gin router
	router := gin.Default()

	// Configure CORS
	corsOrigins := os.Getenv("CORS_ALLOWED_ORIGINS")
	if corsOrigins == "" {
		corsOrigins = "http://localhost:3001"
	}

	router.Use(cors.New(cors.Config{
		AllowOrigins:     strings.Split(corsOrigins, ","),
		AllowMethods:     []string{"GET", "POST", "PUT", "DELETE", "OPTIONS"},
		AllowHeaders:     []string{"Origin", "Content-Type", "Accept", "Authorization"},
		ExposeHeaders:    []string{"Content-Length"},
		AllowCredentials: true,
		MaxAge:           12 * 60 * 60, // 12 hours
	}))

	// Root endpoint
	router.GET("/", func(c *gin.Context) {
		c.JSON(http.StatusOK, gin.H{
			"message": "Personal MD Files Backend",
		})
	})

	// Chat endpoint
	router.POST("/api/chat", chatController.Chat)

	// Providers endpoint
	router.GET("/api/providers", func(c *gin.Context) {
		providers := chatController.Providers()
		c.JSON(http.StatusOK, gin.H{"providers": providers})
	})

	// Switch provider endpoint
	router.POST("/api/switch-provider", chatController.SwitchProvider)

	// Start the server
	port := os.Getenv("PORT")
	if port == "" {
		port = "8008"
	}

	log.Printf("Starting server on port %s", port)
	if err := router.Run(fmt.Sprintf(":%s", port)); err != nil {
		log.Fatalf("Failed to start server: %v", err)
	}
}
